admin/load.php

```diff
@@ -503,3 +533,37 @@ function perflab_dismiss_wp_pointer_wrapper() {
 	check_ajax_referer( 'dismiss_pointer' );
 }
 add_action( 'wp_ajax_dismiss-wp-pointer', 'perflab_dismiss_wp_pointer_wrapper', 0 );
+
+/**
+ * Callback function hooked to admin_init to handle WPP inline standalone
+ * plugin actions such as activation and deactivation.
+ *
+ * @since n.e.x.t
+ * @return void
+ */
+function perflab_handle_inline_plugin_actions() {
+	if (
+		! empty( filter_input( INPUT_GET, 'wpp' ) ) &&
+		! empty( filter_input( INPUT_GET, 'action' ) ) &&
+		! empty( filter_input( INPUT_GET, 'plugin' ) ) &&
+		'1' === filter_input( INPUT_GET, 'wpp' ) &&
+		(
+			'deactivate' === filter_input( INPUT_GET, 'action' ) ||
+			'activate' === filter_input( INPUT_GET, 'action' )
+		)
+	) {
+		$referer = wp_get_referer();
```

Comment:
This function is missing capability checks, which would be a security flaw.

We must add similar checks here that protect plugin activation and deactivation in core:
* For plugin activation, check `current_user_can( 'activate_plugin', $plugin )`.
* For plugin deactivation, check `current_user_can( 'deactivate_plugin', $plugin )`.

For the case that those checks aren't satisfied, you can use the same approach applied in core, even with the same translation message (saves us two custom translation strings when it's not needed). See the WP core `wp-admin/plugins.php` file for reference.

Additionally, this function is also missing the nonce check, which in combination with the above would be a more severe security problem. You should be able to rely on calling `check_admin_refer()` at the beginning of the function, with the action used below when creating the `_wpnonce` parameter in the URL (e.g. `activate-plugin_{$plugin}` or `deactivate-plugin_{$plugin}`).

Reply:
@felixarntz this has been addressed in the following commits:

- https://github.com/WordPress/performance/pull/864/commits/bb58fc5982cebc657dc234313a52e2aa55ed51e7
- https://github.com/WordPress/performance/pull/864/commits/ddaf667b43e9519612466ed7a9269c55fb1dcd1c
- https://github.com/WordPress/performance/pull/864/commits/2583f3d0581bc348e1d8f0c6e3c4ff2dc511b2d9